src/exec-child.js

```diff
@@ -18,29 +18,32 @@ var execOptions = params.execOptions;
 var pipe = params.pipe;
 var stdoutFile = params.stdoutFile;
 var stderrFile = params.stderrFile;
-var codeFile = params.codeFile;
-
-var c = childProcess.exec(cmd, execOptions, function (err) {
-  if (!err) {
-    fs.writeFileSync(codeFile, '0');
-  } else if (err.code === undefined) {
-    fs.writeFileSync(codeFile, '1');
-  } else {
-    fs.writeFileSync(codeFile, err.code.toString());
-  }
-});
+
+var c;
+try {
+  c = childProcess.exec(cmd, execOptions, function (err) {
+    if (!err) {
+      process.exitCode = 0;
+    } else if (err.code === undefined) {
+      process.exitCode = 1;
+    } else {
+      process.exitCode = err.code;
+    }
+  });
+} catch (e) {
+  // child_process could not run the command.
+  /* istanbul ignore next */
+  process.exitCode = 127;
+}
```

Comment:
If `childProcess.exec` throws, then `c` will be `undefined`, right? That means the code below this try/catch will fail.

Reply:
I think you're right. We should actually think of valid cases which might trigger this and document them. I think `maxBuffer` normally triggers this, but we're hitting the `maxBuffer` on the `execFileSync()` call in `exec.js` first.

Reply:
Looking at the current code for [`childProcess.exec`](https://github.com/nodejs/node/blob/master/lib/child_process.js#L137), I'm not sure it can throw synchronous exceptions. Not sure about past versions of node, however.

Reply:
Hmmm you may be right.

I tried adding all sorts of broken/invalid cases and couldn't actually trigger this.

Reply:
Removed it.

Reply:
Just for the sake of verifying, I looked through the history of `childProcess.exec` and it hasn't changed much in 2 years, basically since `v0.12`. So, it is safe to remove the `try/catch`